samples/gibbs_ensemble/create_fits.py

```diff
@@ -0,0 +1,197 @@
+"""
+Take in all simulation data, compute equilibrium values for every datafile and take means for simulations with same temperature;
+Plot the phase diagram and the fitted critical point.
+"""
+
+import numpy as np
+from scipy.optimize import curve_fit
+from scipy import signal
+import pickle
+import gzip
+import matplotlib.pyplot as plt
+import argparse
+
+parser = argparse.ArgumentParser(
+    description="Computes mean values for every file given as input, computes the mean values for same temperatures and different seeds, tries to compute the critical point using rectilinear and scaling law and plots the phase diagram.")
+parser.add_argument(
+    'files',
+    nargs='+',
+    help="(All) files of the simulated system")
+args = parser.parse_args()
+
+
+# Warmup length to skip at beginning
+WARMUP_LENGTH = int(3e5)
+
+# Lists for (unsorted) densities and temperature
+dens_liquid = []
+dens_gas = []
+errors_liquid = []
+errors_gas = []
+kTs = []
+
+# 3D critical exponent (See Frenkel, Smits: "Understanding Molecular
+# simulation", p.217
+beta = 0.32
+
+
+# Autocorrelation function to compute the error
+def autocorr(x):
+    _x = x - np.mean(x)
+    return (signal.convolve(
+        _x, _x[::-1], mode='full', method='auto')[(np.size(_x) - 1):]) / (np.sum(_x * _x))
+
+
+def exp(x, a):
+    return np.exp(-a * x)
+
+
+def get_relaxation_time(x):
+    corr = autocorr(x)
+    popt, _ = curve_fit(exp, range(np.size(corr)), corr, p0=(1. / 15000))
+    return popt[0]
+
+
+def compute_error(x):
+    autocorr_time = get_relaxation_time(x)
+    N_eff = np.size(x) / (2 * autocorr_time)
+    return np.sqrt(np.var(x) / N_eff)
+
+
+# Scaling law and Rectilinear law as shown in Frenkel, Smit, eqs. (8.3.6)
+# and (8.3.7)
+def scaling_law_fit(kT, B, kT_c):
+    return B * np.abs(kT - kT_c)**beta
+
+
+def rectilinear_law_fit(kT, p_c, A, kT_c):
+    return p_c + A * np.abs(kT - kT_c)
+
+
+# Load data
+for f in args.files:
```

Comment:
instead of global variables and module level statements consider using functions and return the results

Reply:
I find it more clear to extract and store all data from the files in one block rather than outsourcing it to functions, please leave a suggestion how this could be handled better.

Reply:
In a larger software the single file extraction should definitely be in a function. For a plotting script, I'd say this is OK.